blockchain/blockchain.go

```diff
@@ -98,7 +98,7 @@ func NewProcessor(boltDB *bolt.DB, metadata *ServiceMetadata) (Processor, error)
 			p.address = crypto.PubkeyToAddress(p.privateKey.PublicKey).Hex()
 		}
 	}
-
+	defer ethereumClient.Close()
```

Comment:
Here ```ethereumClient``` should be open till ```Processor``` object is not needed. Right way to do it is adding ```Close()``` method to ```Processor``` and call it in ```Close()``` of ```components.go```. But for this PR at least I think we can leave it as is as when process is finished OS will close the socket.